pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarSession.java

```diff
@@ -958,7 +967,9 @@ void setupConnectionConsumerTask(List<Message> foreignMessages, Runnable postExe
   @Override
   public PulsarMessageProducer createProducer(Destination destination) throws JMSException {
     connection.setAllowSetClientId(false);
-    return new PulsarMessageProducer(this, destination);
+    return getFactory().isUseTemporaryProducers()
```

Comment:
we can have a field and not call that method from getFactory() because it is `synchronized` and we wlll introduce risks for deadlocks